app/models/concerns/applicant/nir.rb

```diff
@@ -3,22 +3,33 @@ module Applicant::Nir
 
   included do
     validate :nir_is_valid, if: :nir?
+    after_validation :add_nir_key
```

Comment:
Est-ce que ce serait pas plus logique d'ajouter la clé avant la validation ? 
On peut ajouter la condition sur la présence du nir aussi 
```suggestion
    before_validation :add_nir_key, if: :nir?
    validate :nir_is_valid, if: :nir?
```

Reply:
C'est d'autant plus important parce qu'il se peut que la formule de la méthode `nir_key` donne un seul chiffre si les 13 premier chiffre sont invalides. Le fait de l'ajouter avant validation permettrait de détecter ces cas.